agent/iskakalunan.py

```diff
@@ -0,0 +1,350 @@
+
+logger = logging.getLogger('iskakalunan')
+logging.basicConfig(format='%(module)s %(levelname)s %(message)s', level=logging.DEBUG)
+
+
+class Iskakalunan(object):
+    def __init__(self):
+        """Basic Iskakalunan model
+
+        messages: mail queue on g-mail
+        """
+        self.credentials = self.get_credentials()
+        http = self.credentials.authorize(httplib2.Http())
+        self.service = discovery.build('gmail', 'v1', http=http)
+
+    def fetch_messages(self, label_id):
+        """Fetch messages from pre-set mail account
+
+        valid results will be updated to messages
+            - Valid result means messages contain 'Iskakalunan' (case insensitive)
+            - Only messages tags w/ 'NewRequest', 'InQueue', and 'InProcess'
+        """
+        try:
+            response = self.service.users().messages().list(userId='me',
+                                                       labelIds=label_id).execute()
+            message_list = []
+            messages = []
+            if 'messages' in response:
+                message_list.extend(response['messages'])
+
+            while 'nextPageToken' in response:
+                page_token = response['nextPageToken']
+                response = service.users().messages().list(userId=user_id,
+                                                           labelIds=label_id,
+                                                           pageToken=page_token).execute()
+                message_list.extend(response['messages'])
+
+            for message in message_list:
+                message_temp = Message(self.service, message['id'], message['threadId'])
+                if not 'moztpeqa' in message_temp.sender:
+                    logger.debug("message sender: %s" % message_temp.sender)
+                    messages.append(message_temp)
+
+            return messages
+        except errors.HttpError, error:
+            logger.error("An error occurred: %s" % error)
+
+    def prepare_try_build(self, revision):
+        url_try_builds = "https://archive.mozilla.org/pub/firefox/try-builds/"
+        req = urllib2.Request(url_try_builds)
+        response = urllib2.urlopen(req)
+        html_try_builds = response.read()
+
+        pattern = 'href="(.+' + message.revision + '\/)"'
+        m = re.search(pattern, html_try_builds)
+        url_revision_build = "https://archive.mozilla.org" + m.group(1) + "try-win64/firefox-55.0a1.en-US.win64.zip"
+        logger.debug("Download build from %s" % url_revision_build)
+        file_revision_build = urllib2.urlopen(url_revision_build)
+        with open('firefox.zip', 'wb') as output:
+            while True:
+                data = file_revision_build.read(4096)
+                if data:
+                    output.write(data)
+                else:
+                    break
+
+        with zipfile.ZipFile('firefox.zip', 'r') as zf:
+            zf.extractall()
+        shutil.rmtree(r'C:/Program Files/Mozilla Firefox')
```

Comment:
You might need to check the local PATH environment variable, some machine will set the firefox path to C:/Program Files (x86)/Mozilla Firefox...